docs/spec.md

```diff
@@ -9,6 +9,27 @@ for other tools.
 The core Git LFS idea is that instead of writing large blobs to a Git repository,
 only a pointer file is written.
 
+* Pointer files are text files which MUST contain only UTF-8 characters.
+* Each line MUST be of the format `{key} {value}\n` (trailing unix newline).
+* Only a single space character between `{key}` and `{value}`.
+* Keys MUST only use the characters `[a-z] [0-9] . -`.  
+* The first key is _always_ `version`.
+* Lines of key/value pairs MUST be sorted alphabetically in ascending order
+(with the exception of `version`, which is always first).
+* Values MUST NOT contain return or newline characters.
+* Pointer files SHOULD NOT have the executable bit set when checked-in in Git.
+
+The required keys are:
+
+* `version` is a URL that identifies the pointer file spec.  Parsers MUST use
+simple string comparison on the version, without any URL parsing or
+normalization.  It is case sensitive, and %-encoding is discouraged.
+* `oid` tracks the unique object id for the file, prefixed by its hashing
```

Comment:
The spec should be 100% self contained: examples are not the spec.

I agree the column needs to be mentioned here.
On Apr 20, 2015 12:39 PM, "David Ginsburg" <notifications@github.com> wrote:

> In docs/spec.md
> <https://github.com/github/git-lfs/pull/246#discussion_r28721411>:
>
> > +* Pointer files are text files which MUST contain only UTF-8 characters.
> > +* Each line MUST be of the format `{key} {value}\n` (trailing unix newline).
> > +* Only a single space character between `{key}` and `{value}`.
> > +* Keys MUST only use the characters `[a-z] [0-9] . -`.
> > +* The first key is _always_ `version`.
> > +* Lines of key/value pairs MUST be sorted alphabetically in ascending order
> > +(with the exception of `version`, which is always first).
> > +* Values MUST NOT contain return or newline characters.
> > +* Pointer files SHOULD NOT have the executable bit set when checked-in in Git.
> > +
> > +The required keys are:
> > +
> > +* `version` is a URL that identifies the pointer file spec.  Parsers MUST use
> > +simple string comparison on the version, without any URL parsing or
> > +normalization.  It is case sensitive, and %-encoding is discouraged.
> > +* `oid` tracks the unique object id for the file, prefixed by its hashing
>
> Do you want to be explicit about the prefix being followed by a ":" or is
> that clear from the example?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/github/git-lfs/pull/246/files#r28721411>.
>